### PR TITLE
Fix Issue #3325

### LINKED ISSUE #3325: [Documentation]: config.toml options should be documented on the doc web site
**What problem or use case are you trying to solve?**

It seems that there is no documentation on all the options available to configure OpenDevin on the documentation web site: https://github.com/All-Hands-AI/OpenHands/tree/main/docs

**Describe the UX of the solution you'd like**

It would be ideal if we could get some documentation on the web site.

**Do you have thoughts on the technical implementation?**

This should be created as its own page, and also added to the appropriate place in sidebars.ts: https://github.com/All-Hands-AI/OpenHands/blob/main/docs/sidebars.ts

To figure out what configuration options are available, we can look at the following file:
https://github.com/All-Hands-AI/OpenHands/blob/1109637efbe72b524473f0a4a4b84bbc74db3764/openhands/core/schema/config.py#L4

### PR DESCRIPTION
**Pull request by Mr-Imperium**
Added a configuration options md file to openhands/docs. Added appropriately to sidebars.ts. Fixes #3325

**End-user friendly description of the problem this fixes or functionality that this introduces**

- [X] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
Resolves #3325

### CODE REVIEW

**Review by enyst — CHANGES_REQUESTED** [review]
Thank you for working on this, @Mr-Imperium ! This issue might be tricky, because of just how many configuration options there are.

- I would mostly suggest to take a look at `config.toml.template` file, [here](https://github.com/All-Hands-AI/OpenHands/blob/main/config.template.toml), we have usually tried to keep it up to date, as an example that users can modify as they need
- I'm afraid the LLM you used hallucinated some default values 😅. For example, AWS defaults are all None, there's no 'us-west' anywhere. The LLM specific options, including AWS options, are [here](https://github.com/All-Hands-AI/OpenHands/blob/01cacf7c331b29b915c970561aeffd41dff55def/openhands/core/config/llm_config.py#L16), the same for others, like prompt caching is by default `True`.
- for the record, the [config module](https://github.com/All-Hands-AI/OpenHands/tree/main/openhands/core/config) where all *config.py files are, is the implementation of all these options

**Review by Monkub81 — APPROVED** [review]

**Review by Monkub81 — APPROVED** [review]

**Comment by enyst** [comment]
@Mr-Imperium Sorry, I think there are two files now? Can you please take a look

**Comment by Mr-Imperium** [comment]
> @Mr-Imperium Sorry, I think there are two files now? Can you please take a look

Updated, please review them.

File: docs/modules/usage/configuration-options.md [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+- `disable_color`
+  - Type: `bool`
+  - Default: `false`
+  - Description: Disable color in terminal output
+
+**Session Management**
+- `enable_cli_session`
+  - Type: `bool`
+  - Default: `false`
+  - Description: Enable saving and restoring the session when run from CLI
```
Comment (enyst) [inline docs/modules/usage/configuration-options.md:99]: Just a small thing here, this does not exist anymore. It was removed sometime last month or so

File: docs/modules/usage/configuration-options.md [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+  - Description: Runtime environment
+
+- `default_agent`
+  - Type: `str`
+  - Default: `"CodeActAgent"`
+  - Description: Name of the default agent
+
+- `jwt_secret`
+  - Type: `str`
+  - Default: `""`
```
Comment (enyst) [inline docs/modules/usage/configuration-options.md:183]:
```suggestion
  - Default: `uuid.uuid4().hex`
```

File: docs/modules/usage/configuration-options.md [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+  - Description: JWT secret for authentication
```
Comment (enyst) [inline docs/modules/usage/configuration-options.md:184]:
```suggestion
  - Description: JWT secret for authentication. Please set it to your own value.
```

File: docs/modules/usage/configuration-options.md [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+
+- `workspace_mount_path`
+  - Type: `str`
+  - Default: `""`
+  - Description: Path to mount the workspace
+
+- `workspace_mount_rewrite`
+  - Type: `str`
+  - Default: `""`
+  - Description: Path to rewrite the workspace mount path to
```
Comment (enyst) [inline docs/modules/usage/configuration-options.md:163]:
```suggestion
  - Description: Path to rewrite the workspace mount path to. You can usually ignore this, it refers to special cases of running inside another container.
```

File: docs/modules/usage/configuration-options.md [hunk 5]
```diff
@@ -0,0 +1,465 @@
...
+- `enable_cli_session`
+  - Type: `bool`
+  - Default: `false`
+  - Description: Enable saving and restoring the session when run from CLI
+
+**Trajectories**
+- `trajectories_path`
+  - Type: `str`
+  - Default: `"./trajectories"`
+  - Description: Path to store trajectories (can be a folder or a file)
```
Comment (enyst) [inline docs/modules/usage/configuration-options.md:105]:
```suggestion
  - Description: Path to store trajectories (can be a folder or a file). If it's a folder, the trajectories will be saved in a file named with the session id name and .json extension, in that folder.
```

File: docs/modules/usage/configuration-options.md [hunk 6]
```diff
@@ -0,0 +1,465 @@
...
+
+- `aws_secret_access_key`
+  - Type: `str`
+  - Default: `""`
+  - Description: AWS secret access key
+
+**API Configuration**
+- `api_key`
+  - Type: `str`
+  - Default: `"your-api-key"`
```
Comment (enyst) [inline docs/modules/usage/configuration-options.md:208]:
```suggestion
  - Default: `None`
```

File: docs/modules/usage/configuration-options.md [hunk 7]
```diff
@@ -0,0 +1,465 @@
...
+
+- `embedding_model`
+  - Type: `str`
+  - Default: `"local"`
+  - Description: Embedding model to use
+
+**Message Handling**
+- `max_message_chars`
+  - Type: `int`
+  - Default: `10000`
```
Comment (enyst) [inline docs/modules/usage/configuration-options.md:256]:
```suggestion
  - Default: `30000`
```

File: docs/modules/usage/configuration-options.md [hunk 8]
```diff
@@ -0,0 +1,465 @@
...
+  - Description: Maximum number of characters in an observation's content
```
Comment (enyst) [inline docs/modules/usage/configuration-options.md:257]:
```suggestion
  - Description: The approximate maximum number of characters in the content of an event included in the prompt to the LLM. Larger observations are truncated.
```

File: docs/modules/usage/configuration-options.md [hunk 9]
```diff
@@ -0,0 +1,465 @@
...
+  - Description: Timeout for the API
+
+- `top_p`
+  - Type: `float`
+  - Default: `1.0`
+  - Description: Top p for the API
+
+- `disable_vision`
+  - Type: `bool`
+  - Default: `true`
```
Comment (enyst) [inline docs/modules/usage/configuration-options.md:329]:
```suggestion
  - Default: `None`
```

**Comment by enyst** [comment]
@Mr-Imperium This looks good! Only had very minor tweaks to suggest, please take a quick look if you can.

Happy to approve this PR if that's fine with you. This is a good start to document all these, thank you for doing this!

**Comment by Mr-Imperium** [comment]
> @Mr-Imperium This looks good! Only had very minor tweaks to suggest, please take a quick look if you can.
> 
> Happy to approve this PR if that's fine with you. This is a good start to document all these, thank you for doing this!

Happy to have been able to contribute! Just a beginner; your guidance has been invaluable. Thanks!

File: docs/modules/usage/configuration-options.md [hunk 10]
```diff
@@ -0,0 +1,459 @@
+---
+id: configuration-options
+title: Configuration Options
+sidebar_label: Configuration Options
+---
+
+# Configuration Options
+
+This guide details all configuration options available for OpenHands, helping you customize its behavior and integrate it with other services. Required options and those with recommended settings are highlighted.
```
Comment (enyst) [inline docs/modules/usage/configuration-options.md:9]:
```suggestion
This guide details all configuration options available for OpenHands, helping you customize its behavior and integrate it with other services.
```

**Review by enyst — APPROVED** [review]

**Comment by enyst** [comment]
@mamoodi I think the contents here are great to have and this kind of doc is needed for a long time, so I'd like to take this in.

**Pull request merged**